docs/getting-started.md

```diff
@@ -0,0 +1,140 @@
+also, thanks to gazelle addon, there is a rule for building each target on its own.
+
+#### Install 
+Prerequisites:
+python
+
+https://docs.bazel.build/versions/master/install.html
+cd cluster-provision/gocli
+
+Using local gocli images before publishing
+
+```
+make container-run
+export KUBEVIRTCI_GOCLI_CONTAINER=bazel:gocli
+```
+
+Publishing (after make container-run / make all)
+```
+make push 
+```
+and then update cluster-up/cluster/images.sh with the gocli hash.
+
+# Provisioning (ocp/okd as example)
+
+Background:
+Create 2 containers, one of the gocli, one is the actual cluster container.
+
+The gocli container mounts the docker socket, so it can interact with the 2nd container and run commands there.
+
+Create your pull secret, save as pull-secret file.
+https://cloud.redhat.com/openshift/install/metal/user-provisioned
+Sometimes registry.svc.ci.openshift.org auth need to be added to it, get it internally.
+
+```
+export INSTALLER_PULL_SECRET=$(realpath pull-secret)
+./cluster-provision/ocp/4.3/provision.sh
+```
+
+Test a provider locally before publishing 
+```
+export KUBEVIRTCI_PROVISION_CHECK=1
+export KUBEVIRT_PROVIDER=ocp-4.3
+make cluster-up
+```
+
+Publish 
+
+Prerequisites:
+you should have kubevirtci quay write access. (or docker.io for okd)
+```
+docker login -u="user" -p="pass" quay.io
+./cluster-provision/ocp/4.3/publish.sh
+```
+
+# Creating new provider
+Clone folder of ocp, folder name should be x/y as in the provider name x-y (ie. ocp-4.3)\
+cluster-provision/ocp/4.3/provision.sh  # used to create a new provider)\
+cluster-provision/ocp/4.3/publish.sh  # used to publish new provider)\
+cluster-provision/ocp/4.3/run.sh\
+cluster-provision/ocp/README.md\
+cluster-up/cluster/ocp-4.3/README.md\
+cluster-up/cluster/ocp-4.3/provider.sh  # used by cluster-up
+
+# Flow of running provision.sh
+cluster-provision/ocp/4.3/provision.sh 
+
+Run gocli okd provision with parameters such as:\
+ocp release image (ie registry.svc.ci.openshift.org/ocp/release:4.3)\
+secret file (in order to pull images)\
+memory size\
+scripts, manifests, hacks directories that will be accessible from both of the containers (gocli and cluster)
+
+cluster-provision/gocli/cmd/okd/provision.go\
+Parse arguments, validate them, mount the 3 folders so they would be accessible from the cluster container.\
+Pull ocp base image.\
+Run scripts/provision.sh in the cluster container
+
+cluster-provision/okd/scripts/provision.sh\
+compiles openshift-installer, copied it to the container, and runs it.\
+delete secrets.\
+shutdown vms (so it will be ready to docker commit, stable status).\
+
+container commit (creates an image out of the container) in case all succeed.
+
+# Flow of running make cluster-up
+./cluster-up/cluster/ocp-4.3/provider.sh\
+parse arguments of bash, for example extract pull secret.\
+runs gocli okd run (./cluster-provision/gocli/cmd/okd/run.go according Cobra path)
+
+./cluster-provision/gocli/cmd/okd/run.go\
```

Comment:
Maybe it would be better to use a list instead of enforcing newlines. It would be definitely easier to read.